api/konnect/v1alpha2/konnect_gateway_controlplane_types.go

```diff
@@ -0,0 +1,285 @@
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:default={{type: "Programmed", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// GetKonnectLabels gets the Konnect Labels from object's spec.
+func (c *KonnectGatewayControlPlane) GetKonnectLabels() map[string]string {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		return nil
+	}
+
+	return c.Spec.CreateControlPlaneRequest.Labels
+}
+
+// SetKonnectLabels sets the Konnect Labels in object's spec.
+func (c *KonnectGatewayControlPlane) SetKonnectLabels(labels map[string]string) {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		c.Spec.CreateControlPlaneRequest = &sdkkonnectcomp.CreateControlPlaneRequest{
+			Labels: labels,
+		}
+	} else {
+		c.Spec.CreateControlPlaneRequest.Labels = labels
+	}
+}
+
+// GetKonnectName gets the Name from CreateControlPlaneRequest.
+func (c *KonnectGatewayControlPlane) GetKonnectName() string {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		return ""
+	}
+	return c.Spec.CreateControlPlaneRequest.Name
+}
+
+// SetKonnectName sets the Name in CreateControlPlaneRequest.
+func (c *KonnectGatewayControlPlane) SetKonnectName(name string) {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		c.Spec.CreateControlPlaneRequest = &sdkkonnectcomp.CreateControlPlaneRequest{
+			Name: name,
+		}
+	} else {
+		c.Spec.CreateControlPlaneRequest.Name = name
+	}
+}
+
+// GetKonnectClusterType gets the ClusterType from CreateControlPlaneRequest.
+func (c *KonnectGatewayControlPlane) GetKonnectClusterType() *sdkkonnectcomp.CreateControlPlaneRequestClusterType {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		return nil
+	}
+	return c.Spec.CreateControlPlaneRequest.ClusterType
+}
+
+// SetKonnectClusterType sets the ClusterType in CreateControlPlaneRequest.
+func (c *KonnectGatewayControlPlane) SetKonnectClusterType(clusterType *sdkkonnectcomp.CreateControlPlaneRequestClusterType) {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		c.Spec.CreateControlPlaneRequest = &sdkkonnectcomp.CreateControlPlaneRequest{
+			ClusterType: clusterType,
+		}
+	} else {
+		c.Spec.CreateControlPlaneRequest.ClusterType = clusterType
+	}
+}
+
+// GetKonnectCloudGateway gets the CloudGateway from CreateControlPlaneRequest.
+func (c *KonnectGatewayControlPlane) GetKonnectCloudGateway() *bool {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		return nil
+	}
+	return c.Spec.CreateControlPlaneRequest.CloudGateway
+}
+
+// SetKonnectCloudGateway sets the CloudGateway in CreateControlPlaneRequest.
+func (c *KonnectGatewayControlPlane) SetKonnectCloudGateway(cloudGateway *bool) {
+	if c.Spec.CreateControlPlaneRequest == nil {
+		c.Spec.CreateControlPlaneRequest = &sdkkonnectcomp.CreateControlPlaneRequest{
+			CloudGateway: cloudGateway,
+		}
+	} else {
+		c.Spec.CreateControlPlaneRequest.CloudGateway = cloudGateway
+	}
```

Comment:
Nit, take it or leave it:

```suggestion
	if c.Spec.CreateControlPlaneRequest == nil {
		c.Spec.CreateControlPlaneRequest = &sdkkonnectcomp.CreateControlPlaneRequest{}
	}
	c.Spec.CreateControlPlaneRequest.CloudGateway = cloudGateway
```

Reply:
Similarly other funcs that set a field.